### PR TITLE
[iamclient] Subscribe on subjects changed event

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: iamclient/iamclient.go [hunk 1]
```diff
@@ -145,7 +140,13 @@ func New(
 		return nil, aoserrors.Wrap(err)
 	}
 
-	go localClient.connectPublicNodeService()
+	ctx, cancelFunction := context.WithCancel(context.Background())
+
+	localClient.cancelFunc = cancelFunction
+
+	go localClient.connectPublicIdentityService(ctx)
+
```
Comment (al1img) [inline iamclient/iamclient.go:148]: Remove empty line

File: iamclient/iamclient.go [hunk 2]
```diff
@@ -47,8 +47,9 @@ import (
  **********************************************************************************************************************/
 
 const (
-	iamRequestTimeout    = 30 * time.Second
-	iamReconnectInterval = 10 * time.Second
+	iamRequestTimeout      = 30 * time.Second
+	iamReconnectInterval   = 10 * time.Second
+	disconnectedFromIAMLog = "Disconnected from IAM"
```
Comment (al1img) [inline iamclient/iamclient.go:52]: No need to put logs as const. Either suppress linter or better to rephrase in different use cases.

File: iamclient/iamclient.go [hunk 3]
```diff
@@ -160,6 +161,23 @@ func (client *Client) GetSystemID() (systemID string) {
 	return client.systemID
 }
 
+// GetSubjects returns subjects.
+func (client *Client) GetSubjects() (subjects []string, err error) {
```
Comment (al1img) [inline iamclient/iamclient.go:165]: GetUnitSubjects

File: iamclient/iamclient.go [hunk 4]
```diff
@@ -210,6 +228,19 @@ func (client *Client) SubscribeNodeInfoChange() <-chan cloudprotocol.NodeInfo {
 	return ch
 }
 
+// SubscribeSubjectsChanged subscribes client on subject changed events.
+func (client *Client) SubscribeSubjectsChanged() <-chan []string {
```
Comment (al1img) [inline iamclient/iamclient.go:232]: SubscribeUnitSubjectsChanged

File: unitstatushandler/unitstatushandler.go [hunk 5]
```diff
@@ -41,6 +41,12 @@ import (
  * Types
  **********************************************************************************************************************/
 
+// UnitInfoProvider provides information about the unit.
+type UnitInfoProvider interface {
```
Comment (al1img) [inline unitstatushandler/unitstatushandler.go:45]: Consider renaming `NodeManager` -> `UnitManager` and extend its API.

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

File: iamclient/iamclient.go [hunk 6]
```diff
@@ -640,6 +724,24 @@ func (client *Client) connectPublicNodeService() {
...
+	listener pb.IAMPublicIdentityService_SubscribeSubjectsChangedClient, err error,
+) {
+	client.Lock()
+	defer client.Unlock()
+
+	client.identService = pb.NewIAMPublicIdentityServiceClient(client.publicConnection)
+
+	listener, err = client.identService.SubscribeSubjectsChanged(context.Background(), &emptypb.Empty{})
+	if err != nil {
+		log.WithField("error", err).Error("Can't subscribe on unit subjects change event")
```
Comment (al1img) [inline iamclient/iamclient.go:737]: Keep in go projects displaying err as `Can't subscribe on unit subjects change event: %v", err`

File: iamclient/iamclient_test.go [hunk 7]
```diff
@@ -165,6 +172,65 @@ func TestGetSystemID(t *testing.T) {
 	}
 }
 
+func TestGetUnitSubjects(t *testing.T) {
+	publicServer, protectedServer, err := newTestServer(publicServerURL, protectedServerURL)
+	if err != nil {
+		t.Fatalf("Can't create test server: %s", err)
```
Comment (al1img) [inline iamclient/iamclient_test.go:178]: %v for error. Here and below.

**Review by al1img — APPROVED** [review]

**Pull request merged**